internal/backends/python/gen_pypi_map/package_index.go

```diff
@@ -26,9 +27,8 @@ func FakePackageIndex(packages ...string) PackageIndex {
 }
 
 func NewPackageIndex(index string, limit int) (PackageIndex, error) {
-	resp, err := http.Get(index)
-
-	if err != nil {
+	resp, err := api.HttpClient.Get(index)
+	if resp == nil || err != nil {
```

Comment:
Why do we have to check for a nil resp here but not in most of the other places we use Get?

Reply:
good question. fixing

Reply:
i think the docs are indicating that `resp` can't be `nil` if `err` is non-`nil`. I'll just do a check in `HttpClient.Do` to assert that that's true, and if it isn't then panic (since without the check it would probably be a segfault due to accessing `nil` anyways)